crates/services/block_aggregator_api/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    api::{
+        BlockAggregatorApi,
+        BlockAggregatorQuery,
+    },
+    blocks::{
+        Block,
+        BlockSource,
+    },
+    db::BlockAggregatorDB,
+};
+use fuel_core_services::{
+    RunnableTask,
+    StateWatcher,
+    TaskNextAction,
+    stream::BoxStream,
+    try_or_stop,
+};
+use result::Result;
+
+pub mod api;
+pub mod blocks;
+pub mod db;
+pub mod result;
+
+#[cfg(test)]
+mod tests;
+
+// TODO: this doesn't need to limited to the blocks,
+//   but we can change the name later
+pub struct BlockAggregator<Api, DB, Blocks> {
+    query: Api,
+    database: DB,
+    block_source: Blocks,
+}
+
+impl<Api, DB, Blocks> RunnableTask for BlockAggregator<Api, DB, Blocks>
+where
+    Api: BlockAggregatorApi,
+    DB: BlockAggregatorDB,
+    Blocks: BlockSource,
+{
+    async fn run(&mut self, watcher: &mut StateWatcher) -> TaskNextAction {
+        tracing::debug!("BlockAggregator running");
+        tokio::select! {
+            query_res = self.query.await_query() => self.handle_query(query_res).await,
```

Comment:
i think this tokio::select! should be reordered in the following manner -

```
biased;

watcher

next_block

query
```

this way if a query comes in that potentially references the latest block we don't return an error

Reply:
Hmm. I think it all depends on how we design the adapters. They don't need to error, for example, just include an incomplete list.

I'm open to adding `bias`, but it's actually been best practice to avoid it. So I prefer leaving it out unless we really want to enforce some order.